lib/storage.js

```diff
@@ -459,9 +459,10 @@ api.remove = async ({edvId, id}) => {
   assert.string(id, 'id');
   _assert128BitId(id);
 
-  const result = await database.collections.edvDoc.deleteOne(
-    {edvId: database.hash(edvId), id: database.hash(id)});
-  return result.result.n !== 0;
+  const result = await database.collections.edvDoc.findOneAndUpdate(
```

Comment:
Why is this API being used instead of `updateOne`? What are the differences? IIRC, there are some special atomic find and modify/update calls that can be performed by mongo but they have very different performance characteristics and constraints (especially with respect to sharding) than the more vanilla calls. I haven't used this particular call before (since it looks like it's part of the new driver API). Can you say more here -- do we need to use this or would `updateOne` suffice?

Reply:
The difference is `findOneAndUpdate` returns the updated document and `updateOne` does not. In this case the document does not need to be returned, so I have updated it to use `updateOne`.

Reply:
We also need to make sure we're gating the update based on the sequence number -- just like the other update call does. Perhaps we can get some reuse out of it ... I'm not sure.

Reply:
This implies a more significant change to the remove API -- it's not enough to simply pass in the identifiers. In light of some of the other discussion around where `deleted` will be tracked -- what do you think would be a good API here for deletion?

Reply:
If we don't want to break the trust and have `deleted` be sent to the server, what about just having the edv-client clear out all content and relevant jwe properties and then send it to the regular `update` route?